pycbc/inference/models/hierarchical.py

```diff
@@ -599,3 +604,316 @@ def _loglikelihood(self):
+        self.primary_model = self.submodels[kwargs['primary_lbl'][0]]
+        self.other_models = self.submodels.copy()
+        self.other_models.pop(kwargs['primary_lbl'][0])
+        self.other_models = list(self.other_models.values())
+
+    def write_metadata(self, fp, group=None):
+        """Adds metadata to the output files
+
+        Parameters
+        ----------
+        fp : pycbc.inference.io.BaseInferenceFile instance
+            The inference file to write to.
+        group : str, optional
+            If provided, the metadata will be written to the attrs specified
+            by group, i.e., to ``fp[group].attrs``. Otherwise, metadata is
+            written to the top-level attrs (``fp.attrs``).
+        """
+        super().write_metadata(fp, group=group)
+        sampattrs = fp.getattrs(group=fp.samples_group)
+        # if a group is specified, prepend the lognl names with it
+        if group is None or group == '/':
+            prefix = ''
+        else:
+            prefix = group.replace('/', '__')
+            if not prefix.endswith('__'):
+                prefix += '__'
+        try:
+            for lbl, model in self.submodels.items():
+                sampattrs['{}lognl'.format(prefix + '%s__' % lbl)
+                          ] = model.lognl
+        except AttributeError:
+            pass
+
+    def total_loglr(self):
+        r"""Computes the total log likelihood ratio,
+
+        .. math::
+
+            \log \mathcal{L}(\Theta) = \sum_i
+                \left<h_i(\Theta)|d_i\right> -
+                \frac{1}{2}\left<h_i(\Theta)|h_i(\Theta)\right>,
+
+        at the current parameter values :math:`\Theta`.
+
+        Returns
+        -------
+        float
+            The value of the log likelihood ratio.
+        """
+        # calculate <d-h|d-h> = <h|h> - 2<h|d> + <d|d> up to a constant
+
+        # note that for SOBHB signals, ground-based detectors dominant SNR
+        # and accuracy of (tc, ra, dec)
+        self.primary_model.return_sh_hh = True
+        sh_primary, hh_primary = self.primary_model.loglr
+        self.primary_model.return_sh_hh = False
+
+        nums = self.primary_model.vsamples
+        margin_params = self.primary_model.marginalize_vector_params.copy()
+        margin_params.pop('logw_partial')
+
+        # add likelihood contribution from space-borne detectors, we
+        # calculate sh/hh for each marginalized parameter point
+        sh_others = numpy.full(nums, 0 + 0.0j)
+        hh_others = numpy.zeros(nums)
+
+        for _, other_model in enumerate(self.other_models):
+            current_params_other = other_model.current_params.copy()
+            for i in range(nums):
+                current_params_other.update(
+                    {key: value[i] for key, value in margin_params.items()})
+                for p in self.primary_model.static_params.keys():
+                    # don't pass primary model's 'f_lower' to LISA
```

Comment:
Why are you copying any static params to LISA? Shouldn't the normal hierarchical handling already handle static params?

Reply:
@ahnitz If I remember correctly, I did this because I wanted to avoid some of 3G's static parameters, such `f_lower` to be used by LISA, so just overwrite it here to make sure.